## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 458720`
Commit: `fc5cf31938b36b1c73407536c91e07c077270421`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.boltztrap2</li>
    <li>python312Packages.boltztrap2.dist</li>
    <li>python313Packages.boltztrap2</li>
    <li>python313Packages.boltztrap2.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.boltztrap2</summary>
<pre>E        ACTUAL: array([[-1.650296e-01-6.428551e-15j, -1.171004e-01-1.944773e-14j,
E                1.153646e-02-3.424107e-16j, ..., -7.022013e-06+2.059828e-24j,
E               -1.240296e-06-8.061357e-22j,  6.696246e-07+1.213018e-22j],...
E        DESIRED: array([[-1.650296e-01-6.580596e-16j, -1.171004e-01-1.441497e-15j,
E                1.153646e-02-3.166899e-16j, ..., -1.240296e-06+1.198026e-22j,
E               -7.022013e-06-2.037889e-22j,  6.696246e-07+5.078019e-22j],...

tests/test_fite.py:73: AssertionError
=============================== warnings summary ===============================
&lt;frozen importlib._bootstrap&gt;:488
  &lt;frozen importlib._bootstrap&gt;:488: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 16 from C header, got 96 from PyObject

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_bandlib.py::test_DOS_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_BTPDOS_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_calc_cv_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_fermiintegrals_Si - AssertionError: 
FAILED tests/test_fite.py::test_fitde3D_saved_noder - AssertionError: 
======== 5 failed, 81 passed, 3 skipped, 1 warning in 592.04s (0:09:52) ========</pre>
</details>
<details>
<summary>python313Packages.boltztrap2</summary>
<pre>E        ACTUAL: array([[-1.650296e-01-6.428551e-15j, -1.171004e-01-1.944773e-14j,
E                1.153646e-02-3.424107e-16j, ..., -7.022013e-06+2.059828e-24j,
E               -1.240296e-06-8.061357e-22j,  6.696246e-07+1.213018e-22j],...
E        DESIRED: array([[-1.650296e-01-6.580596e-16j, -1.171004e-01-1.441497e-15j,
E                1.153646e-02-3.166899e-16j, ..., -1.240296e-06+1.198026e-22j,
E               -7.022013e-06-2.037889e-22j,  6.696246e-07+5.078019e-22j],...

tests/test_fite.py:73: AssertionError
=============================== warnings summary ===============================
&lt;frozen importlib._bootstrap&gt;:488
  &lt;frozen importlib._bootstrap&gt;:488: RuntimeWarning: numpy.ndarray size changed, may indicate binary incompatibility. Expected 16 from C header, got 96 from PyObject

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_bandlib.py::test_DOS_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_BTPDOS_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_calc_cv_Si - AssertionError: 
FAILED tests/test_bandlib.py::test_fermiintegrals_Si - AssertionError: 
FAILED tests/test_fite.py::test_fitde3D_saved_noder - AssertionError: 
======== 5 failed, 81 passed, 3 skipped, 1 warning in 590.21s (0:09:50) ========</pre>
</details>
</details>

--- a/pkgs/development/python-modules/boltztrap2/default.nix
+++ b/pkgs/development/python-modules/boltztrap2/default.nix
@@ -1,18 +1,19 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitLab,
+  stdenv,
   spglib,
   numpy,
   scipy,
   matplotlib,
   ase,
   netcdf4,
-  pythonOlder,
   cython,
   cmake,
   setuptools,
   setuptools-scm,
+  pytestCheckHook,
 }:
 
 buildPythonPackage rec {
@@ -26,12 +27,11 @@ buildPythonPackage rec {
     setuptools-scm
   ];
 
-  disabled = pythonOlder "3.5";
-
-  src = fetchPypi {
-    pname = "boltztrap2";
-    inherit version;
-    hash = "sha256-JUIGh/6AF+xYLmF3QN47/A5E9zPKdhO2lhn97giZJ48=";
+  src = fetchFromGitLab {
+    owner = "sousaw";
+    repo = "BoltzTraP2";
+    tag = "v${version}";
+    hash = "sha256-eocstudmgMkuxa94txU8uqIp8HpNEuWQys7WvRRZ4as=";
   };
 
   postPatch = ''
@@ -57,10 +57,28 @@ buildPythonPackage rec {
     netcdf4
   ];
 
-  # pypi release does no include files for tests
-  doCheck = false;
-
   pythonImportsCheck = [ "BoltzTraP2" ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  preInstallCheck = ''
+    tar xf data.tar.xz
+    rm -rf BoltzTraP2
+  '';
+
+  pytestFlags = [ "tests" ];
+
+  disabledTests = lib.optionals (stdenv.system != "x86_64-linux") [
+    # Tests np.load numpy arrays from disk that were, apparently, saved on
+    # x86_64-linux. Then these files are used to compare results of
+    # calculations, which won't work as expected if running on a different
+    # platform.
+    "test_DOS_Si"
+    "test_BTPDOS_Si"
+    "test_calc_cv_Si"
+    "test_fermiintegrals_Si"
+    "test_fitde3D_saved_noder"
+  ];
 
   meta = with lib; {
     description = "Band-structure interpolator and transport coefficient calculator";

